### ISSUE TITLE
[iOS] Can't pause from headphones

### ISSUE THREAD

**Issue by KaiStarkk** [advplyr/audiobookshelf-app#622]
Pressing the pause button causes the app to rewind a few seconds. It seems to register as two presses (pause then play again).

Tested with Aftershokz Aeropex and AcousticSheep sleepphones, both have same problem. 

Sony wh1000mx3 works correctly. 

~iOS 16.3.1~
~iOS 17.1.1~
iOS 17.2.1
~App version 0.9.66~
~App version 0.9.67~
App version 0.9.68
~Server version 2.5.0~
Server version 2.6.0

Please thumbs up if you have this issue, appears to be preferred way of prioritizing issues.

**Comment by cmccambridge** [comment]
I’m also experiencing this in the iOS app with my soundpeats q30 Bluetooth headset. I have to get my phone out to pause a book, which hurts a lot of my favorite audiobook cases- working out, doing chores, etc.

App: 0.9.63-beta
Backend: 2.2.19

**Comment by advplyr** [comment]
I'm not very familiar with iOS since I use Android but on Android it is similar to how you are describing.

**Comment by KaiStarkk** [comment]
> I'm not very familiar with iOS since I use Android but on Android it is similar to how you are describing.

To clarify, on iPhone it is different from android - when you press pause, it rewinds and continued playing. As though the button was pressed twice. 

Is it possible to implement the fix suggested above by handling togglePlayPauseCommand?

**Comment by wnoakes** [comment]
I see this with my iPhone 13 mini on 17.0.3 using Ultimate Ears Boom 3 and Creative Outlier ONE headphones

**Comment by KaiStarkk** [comment]
@ronaldheft  hoping you might be able to share some wisdom on this one? Appears you’ve had some involvement in the iOS player handlers

**Comment by ronaldheft** [comment]
@KaiStarkk Are you experiencing this with a book that is multi-track (multiple files) or single-track (one file)?

I ask, because I know there are definitely still issues with multi-track in the player, especially around winding back the book a few seconds when resuming.

**Comment by KaiStarkk** [comment]
@ronaldheft just tested, it’s happening with both multi track and single track. 

Essentially it’s impossible to pause from headphones.

Any thoughts on the suggestion above about handling  https://developer.apple.com/documentation/mediaplayer/mpremotecommandcenter/1618992-toggleplaypausecommand

**Comment by KaiStarkk** [comment]
On review it would make sense to do a sanity check when pausing and playing also. i.e.

```
        commandCenter.playCommand.isEnabled = true
        commandCenter.playCommand.addTarget { [weak self] event in
            if (self?.isPlaying()) {
                self?.pause()
            } else {
                self?.play(allowSeekBack: true)
            }
            return .success
        }
        
        commandCenter.pauseCommand.isEnabled = true
        commandCenter.pauseCommand.addTarget { [weak self] event in
            if (self?.isPlaying()) {
                self?.pause()
            } else {
                self?.play(allowSeekBack: true)
            }
            return .success
        }

        commandCenter.togglePlayPauseCommand.isEnabled = true
        commandCenter.togglePlayPauseCommand.addTarget { [weak self] event in
            if (self?.isPlaying()) {
                self?.pause()
            } else {
                self?.play(allowSeekBack: true)
            }
            return .success
        }
```

Just need to get myself a machine which can run Xcode and a non-work iPhone to test with.

**Comment by advplyr** [comment]
Fixed in [v0.9.71-beta](https://github.com/advplyr/audiobookshelf-app/releases/tag/v0.9.71-beta)

**Comment by indiefan** [comment]
Experiencing this issue specifically with a pair of Bose Ultra Open Earbuds.

If I pause the book from my phone screen, the headphones can resume play, but they can't pause the book while it's playing. All of their other controls work fine.

Worth noting, the headphones play/pause the audible app without issue.

Also worth noting, my airpods play/pause ABS without issue.

It's only the specific combination of these Bose headphones + Pause in ABS that is breaking.

I tried relaunching the app and also deleting it and reinstalling but neither helped.

Server Version: v2.8.1
App Version: 0.9.73
